src/tests/cli_tests.py

```diff
@@ -1559,6 +1559,19 @@ def test_rnp_single_export(self):
                         'exported packets mismatch')
         return
 
+    def test_rnp_permissive_key_import(self):
+        # Import keys while skipping bad packets, see #1160
+        clear_keyrings()
+        # Try to import  without --permissive option, should fail.
+        ret, _, _ = run_proc(RNPK, ['--homedir', RNPDIR, '--import-keys', data_path('test_key_edge_cases/pubring-malf-cert.pgp')])
+        if ret == 0:
+            raise_err('Imported bad packets without --permissive option set!')
+        # Import with --permissive
+        ret, _, _ = run_proc(RNPK, ['--homedir', RNPDIR, '--import-keys', '--permissive',data_path('test_key_edge_cases/pubring-malf-cert.pgp')])
+        if ret != 0:
+            raise_err('Failed to import keys with --permissive option')
+        return
```

Comment:
I would suggest to add `run_proc(RNPK, ['--homedir', RNPDIR, '--list-keys', '--with-sigs'])` after successful import and check whether output includes correct keys/sigs.

Reply:
When I run --list-keys --with-sigs on `'test_key_edge_cases/pubring-malf-cert.pgp'` it says 4 keys found, but here in the test it compares key count to 7
https://github.com/rnpgp/rnp/pull/1159/files#diff-2b07a33968c6bdb589acc2af6927783bR5526

```
4 keys found

pub   1024/RSA (Encrypt or Sign) 7bc6709b15c23a4a 2017-07-20 [SC]
      e95a3cbf583aa80a2ccc53aa7bc6709b15c23a4a
uid           key0-uid0
sig           7bc6709b15c23a4a 2017-07-20 key0-uid0
uid           key0-uid1
sig           7bc6709b15c23a4a 2017-07-20 key0-uid0
uid           key0-uid2
sig           7bc6709b15c23a4a 2017-07-20 key0-uid0
sub   1024/RSA (Encrypt or Sign) 1ed63ee56fadc34d 2017-07-20 [E]
      e332b27caf4742a11baa677f1ed63ee56fadc34d
sub   1024/DSA 1d7e8a5393c997a8 2017-07-20 [S] [EXPIRED 2017-11-20]
      c5b15209940a7816a7af3fb51d7e8a5393c997a8
sub   1024/RSA (Encrypt or Sign) 8a05b89fad5aded1 2017-07-20 [E]
      5cd46d2a0bd0b8cfe0b130ae8a05b89fad5aded1
```

Reply:
> When I run --list-keys --with-sigs on 'test_key_edge_cases/pubring-malf-cert.pgp' it says 4 keys found

Which commands did you run?

With following commands, and force permissive flag in code, things went as expected, listing 7 keys:
```
rnpkeys --homedir .rnp --import pubring-malf-cert.pgp
rnpkeys --homedir .rnp --list-keys --with-sigs
```

Reply:
nevermind, actually there was other file imported, not pubring-malf-cert.pgp, my mistake